subsys/canbus/isotp/isotp.c

```diff
@@ -41,6 +42,36 @@ NET_BUF_POOL_VAR_DEFINE(isotp_tx_pool, CONFIG_ISOTP_TX_BUF_COUNT,
 
 static void receive_state_machine(struct isotp_recv_ctx *ctx);
 
+static inline struct can_frame isotp_prepare_frame(struct isotp_msg_id *addr)
```

Comment:
Not sure if this is going to be efficient. In C++, we can rely on return value optimization (copy elision), but not in C.
I'm not sure if the inline forces to not copy the struct on return.

Reply:
I'm absolutely not fluent in assembly. I can at least see that it's not inlined, but kept as a separate function call. Not sure if the function copies the struct on return or not.

<details>
<summary>Disassembly for arm-none-eabi</summary>

```asm
08008008 <prepare_frame>:
{
 8008008:	b538      	push	{r3, r4, r5, lr}
 800800a:	460d      	mov	r5, r1
	struct can_frame frame = {
 800800c:	220d      	movs	r2, #13
 800800e:	2100      	movs	r1, #0
{
 8008010:	4604      	mov	r4, r0
	struct can_frame frame = {
 8008012:	3003      	adds	r0, #3
 8008014:	f001 fb65 	bl	80096e2 <memset>
		.flags = (addr->flags & ISOTP_MSG_IDE) != 0 ? CAN_FRAME_IDE : 0,
 8008018:	79a9      	ldrb	r1, [r5, #6]
		.id = addr->ext_id
 800801a:	682b      	ldr	r3, [r5, #0]
	if ((addr->flags & ISOTP_MSG_FDF) != 0) {
 800801c:	0708      	lsls	r0, r1, #28
		.id = addr->ext_id
 800801e:	f3c3 021c 	ubfx	r2, r3, #0, #29
		.flags = (addr->flags & ISOTP_MSG_IDE) != 0 ? CAN_FRAME_IDE : 0,
 8008022:	f3c1 0380 	ubfx	r3, r1, #2, #1
	if ((addr->flags & ISOTP_MSG_FDF) != 0) {
 8008026:	d505      	bpl.n	8008034 <prepare_frame+0x2c>
		if ((addr->flags & ISOTP_MSG_BRS) != 0) {
 8008028:	06c9      	lsls	r1, r1, #27
		frame.flags |= CAN_FRAME_FDF;
 800802a:	bf54      	ite	pl
 800802c:	f043 0304 	orrpl.w	r3, r3, #4
			frame.flags |= CAN_FRAME_BRS;
 8008030:	f043 030c 	orrmi.w	r3, r3, #12
	return frame;
 8008034:	6821      	ldr	r1, [r4, #0]
 8008036:	7163      	strb	r3, [r4, #5]
 8008038:	f362 011c 	bfi	r1, r2, #0, #29
 800803c:	6021      	str	r1, [r4, #0]
}
 800803e:	4620      	mov	r0, r4
 8008040:	bd38      	pop	{r3, r4, r5, pc}
```
</details>

Even if it takes a copy, it's also a matter of code clarity and reduced duplication. So I'd probably keep it as a function call.

Reply:
Done. It's now a void function with pointer argument.